src/sagemaker/tensorflow/tfs.py

```diff
@@ -0,0 +1,150 @@
+
+
+class TFSModel(Model):
+    FRAMEWORK_NAME = 'tfs'
+    LOG_LEVEL_PARAM_NAME = 'SAGEMAKER_TFS_NGINX_LOGLEVEL'
+    LOG_LEVEL_MAP = {
+        logging.DEBUG: 'debug',
+        logging.INFO: 'info',
+        logging.WARNING: 'warn',
+        logging.ERROR: 'error',
+        logging.CRITICAL: 'crit',
+    }
+
+    def __init__(self, model_data, role, image=None, framework_version=TF_VERSION,
+                 container_log_level=None, predictor_cls=TFSPredictor, **kwargs):
+        """Initialize a TFSModel.
+
+       Args:
+           model_data (str): The S3 location of a SageMaker model data ``.tar.gz`` file.
+           role (str): An AWS IAM role (either name or full ARN). The Amazon SageMaker APIs that
+                create Amazon SageMaker endpoints use this role to access model artifacts.
+           image (str): A Docker image URI (default: None). If not specified, a default image for
+                TensorFlow Serving will be used.
+           framework_version (str): Optional. TensorFlow Serving version you want to use.
+           container_log_level (int): Log level to use within the container (default: logging.ERROR).
+                Valid values are defined in the Python logging module.
+           predictor_cls (callable[str, sagemaker.session.Session]): A function to call to create a
+                predictor with an endpoint name and SageMaker ``Session``. If specified, ``deploy()``
+                returns the result of invoking this function on the created endpoint name.
+           **kwargs: Keyword arguments passed to the ``Model`` initializer.
+       """
+        super(TFSModel, self).__init__(model_data=model_data, role=role, image=image,
+                                       predictor_cls=predictor_cls, **kwargs)
+        self._framework_version = framework_version
+        self._container_log_level = container_log_level
+
+    def prepare_container_def(self, instance_type):
+        image = self._get_image_uri(instance_type)
+        env = self._get_container_env()
+        return sagemaker.container_def(image, self.model_data, env)
+
+    def _get_container_env(self):
+        if not self._container_log_level:
+            return self.env
+
+        if self._container_log_level not in TFSModel.LOG_LEVEL_MAP:
+            logging.warning('ignoring invalid container log level: %s', self._container_log_level)
+            return self.env
+
+        env = dict(self.env)
+        env['SAGEMAKER_TFS_NGINX_LOGLEVEL'] = TFSModel.LOG_LEVEL_MAP[self._container_log_level]
+        return env
+
+    def _get_image_uri(self, instance_type):
+        if self.image:
+            return self.image
+
+        # reuse standard image uri function, then strip unwanted python component
+        region_name = self.sagemaker_session.boto_region_name
+        image = create_image_uri(region_name, TFSModel.FRAMEWORK_NAME, instance_type,
+                                 self._framework_version, 'py3')
```

Comment:
How hard is to change create image uri instead of reusing and striping it.

Reply:
lots of fussy behavior in create_image_uri. it would be easy to fall out of sync